[linux-6.6.y] x86/mce/zhaoxin: Enable mcelog to decode PCIE, ZDI/ZPI, and DRAM errors

leoliu-oc opened the pull request:
zhaoxin inclusion
category: bugfix

-------------------

The mcelog cannot decode PCIE, ZDI/ZPI, and DRAM errors in the FFM (Firmware First Mode).
The purpose of this patch is to enable mcelog to decode PCIE, ZDI/ZPI, and DRAM errors that occur on Zhaoxin processors, so that the cause of these errors can be quickly located.

opsiff commented on drivers/acpi/apei/ghes.c:
```diff
@@ -699,14 +699,21 @@ static bool ghes_do_proc(struct ghes *ghes,
 			atomic_notifier_call_chain(&ghes_report_chain, sev, mem_err);
 
-			arch_apei_report_mem_error(sev, mem_err);
+			arch_apei_report_mem_error(sec_sev, mem_err);
```
我对这里不了解。sev = ghes_severity(estatus->error_severity);和sec_sev = ghes_severity(gdata->error_severity);有什么差异,为什么要把前者改成后者，后续新增的代码是为什么又用前者?

leoliu-oc replied:
这里内核的处理是有问题的，即使不加入zhaoxin patch code，这里也应该是使用sec_sev。sev是一个总的severity，是ghes中各个error的severity的最高值，而在arch_apei_report_mem_error时应该是针对各个具体的error。
而后面的代码，使用sev是合理的，所以不需要修改。

Avenger-285714 replied:
> 这里内核的处理是有问题的，即使不加入zhaoxin patch code，这里也应该是使用sec_sev。sev是一个总的severity，是ghes中各个error的severity的最高值，而在arch_apei_report_mem_error时应该是针对各个具体的error。 而后面的代码，使用sev是合理的，所以不需要修改。

这方面您们CPU厂商可能会更专业一些，既然如此建议将这个改动单独拆分成一个提交，并且看起来也可以提给上游

leoliu-oc replied:
新版本中已经拆分出来了，还烦请review了。后续会考虑向上游主线提交。

leoliu-oc committed:
apei: ghes: Modify the GHES severity of mem error in ghes_do_proc

When executing arch_apei_report_mem_error, replace the previous 'sev'with
'sec_sev'. 'sev' represents the overall severity, which is the highest
severity among individual errors in GHES. However, during
arch_apei_report_mem_error, it should be specific to each individual
error.

Signed-off-by: leoliu-oc <leoliu-oc@zhaoxin.com>

leoliu-oc committed:
x86/mce/zhaoxin: Enable mcelog to decode PCIE, ZDI/ZPI, and DRAM errors

zhaoxin inclusion
category: bugfix

-------------------

The mcelog cannot decode PCIE, ZDI/ZPI, and DRAM errors in the FFM
(Firmware First Mode).
The purpose of this patch is to enable mcelog to decode PCIE, ZDI/ZPI, and
DRAM errors that occur on Zhaoxin processors, so that the cause of these
errors can be quickly located.

Signed-off-by: leoliu-oc <leoliu-oc@zhaoxin.com>
